lib/matplotlib/backend_managers.py

```diff
@@ -14,6 +18,169 @@
 from matplotlib.rcsetup import validate_stringlist
 import matplotlib.backend_tools as tools
 
+from matplotlib import is_interactive
+from matplotlib import rcParams
+from matplotlib.figure import Figure
+from matplotlib.backends import get_backend
+
+
+class FigureManagerEvent(object):
+    """Event for when something happens to this figure manager.
+    i.e. the figure it controls gets closed
+
+    Attributes
+    ----------
+    signal : str
+        The name of the signal.
+
+    figure_manager : FigureManager
+        The figure manager that fired the event.
+    """
+    def __init__(self, signal, figure_manager):
+        self.name = signal
+        self.figure_manager = figure_manager
+
+
+class FigureManager(cbook.EventEmitter):
+    """
+    The FigureManager creates and wraps the necessary components to display a
+    figure, namely the Window, FigureCanvas and Toolbar.  It gets used whenever
+    you want the figure in a standalone window.
+
+    Parameters
+    ----------
+    figure : `matplotlib.figure.Figure`
+        The figure to manage.
+
+    num : int
+        The figure number.
+
+    Attributes
+    ----------
+
+    canvas : `matplotlib.backend_bases.FigureCanvasBase`
+        The GUI element on which we draw.
+
+    toolbar : `matplotlib.backend_bases.NavigationToolbar2`
+        The toolbar used for interacting with the figure.
+
+    window : `matplotlib.backend_bases.WindowBase`
+        The window that holds the canvas and toolbar.
+
+    num : int
+        The figure number.
+    """
+    def __init__(self, figure, num):
+        cbook.EventEmitter.__init__(self)
```

Comment:
@tacaswell, regarding your earlier comment about "understanding 'super' was to consider the mro of some classes with nasty MI": to me this doesn't necessarily imply that we should be plastering "super" all over the place; we should be tying hard to avoid "nasty MI".  I'm always worried about the code base getting harder and harder to understand.  Sometimes I'm close to the point of giving up, and saying that as an amateur, I just can't handle it any more.  Many times when I plunge in to try to figure out a problem, it takes too long.  Each fancy trick, each complicated inheritance tree, makes it less likely that a given mpl user will be able to submit a PR fixing a bug, instead of merely reporting the bug.  OK, enough tirade.  I learned to (sort of) like decorators; maybe I will get used to "super" if we end up relying on it.